sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/ManagedSchemaTransformProvider.java

```diff
@@ -202,7 +203,26 @@ Row getConfigurationRow() {
   static Row getRowConfig(ManagedConfig config, Schema transformSchema) {
     // May return an empty row (perhaps the underlying transform doesn't have any required
     // parameters)
-    return YamlUtils.toBeamRow(config.resolveUnderlyingConfig(), transformSchema, false);
+    String yamlConfig = config.resolveUnderlyingConfig();
+    Map<String, Object> configMap = YamlUtils.yamlStringToMap(yamlConfig);
+
+    // The config Row object will be used to build the underlying SchemaTransform.
+    // If a mapping for the SchemaTransform exists, we use it to update parameter names and align
+    // with the underlying config schema
+    Map<String, String> mapping = MAPPINGS.get(config.getTransformIdentifier());
+    if (mapping != null && configMap != null) {
+      Map<String, Object> remappedConfig = new HashMap<>();
+      for (Map.Entry<String, Object> entry : configMap.entrySet()) {
+        String paramName = entry.getKey();
+        if (mapping.containsKey(paramName)) {
```

Comment:
Right now any parameters that aren't recognized are silently dropped. I think we should be at least warning here.

Reply:
The mapping contains parameter names that should be updated. If a parameter is not included in the mapping, we assume its original name is correct.

But agreed some logging would be helpful. I'll add a log showing the final Row configuration used to build the underlying transform

Reply:
P.S. alternatively we can make it mandatory to include all parameter names in the mapping, regardless if they need to be updated or not.

Reply:
Ah got it - I think leaving it is fine, thanks

> But agreed some logging would be helpful. I'll add a log showing the final Row configuration used to build the underlying transform

Sounds good, thank you!